### DISCUSSION TITLE
Removing pre-existing recognizers

### DISCUSSION THREAD

**Discussion by katiejim** [microsoft/presidio#1022] [Q&A] [answered]
Hello!

I am trying to remove some of the pre-defined recognizers as they are causing confusion with custom regex recognizers that are specific to my industry. I have tried using `analyzer.registry.remove_recognizer(au_tfn_recognizer)` to disable the recognizer for AU tax file numbers, but I get the error "NameError: name 'au_tfn_recognizer' is not defined"

The `analyzer.registry.remove_recognizer()` command seems to only work when disabling recognizers that I have made myself. Please can you let me know if I am missing something, or if it simply isn't possible to disable the pre-defined recognizers?

Thanks so much
Kate

**Comment by omri374** [comment, marked as answer]
Hi @katiejim,

The recognizer name is the class name, so to remove this recognizer, the right call would be:
```python
analyzer.registry.remove_recognizer("AuTfnRecognizer")
```

Alternatively, you could start with an empty list of recognizers and add your own:
```python
from presidio_analyzer import AnalyzerEngine, RecognizerRegistry

registry = RecognizerRegistry()
registry.add_recognizer(my_recognizer)
analyzer = AnalyzerEngine(registry=registry)
analyzer.analyze(...) # would only return values detected by my_recognizer
```

**Reply by katiejim** [reply]
@omri374 thank you so much, that's totally solved the problem.

**Comment by MarPeg53** [comment]
Hello everyone and thanks for the support. I have a similar question. When I remove "DateRecognizer" it does not work. What am I doing wrong?

```
titles_recognizer = PatternRecognizer(supported_entity="TITLE",
                                      deny_list=["Mr.","Mrs.","Miss"])
registry = RecognizerRegistry()
registry.load_predefined_recognizers()

registry.add_recognizer(titles_recognizer)
registry.remove_recognizer("DateRecognizer")

analyzer = AnalyzerEngine(registry=registry)

text="His name is Mr. Jones and he is born 3. July"
results = analyzer.analyze(text=text, language="en")
print(results)
```

**Reply by omri374** [reply]
Hi @MarPeg53, Date is special as it is handled by both spaCy and a rule-based recognizer. You can remove DATE_TIME from the supported entities if you're not interested in it at all.

**Reply by MarPeg53** [reply]
@omri374 thank you for your help! Can you tell me how to remove DATE_TIME? Probably not with registry.remove_recognizer("DATE_TIME").

**Reply by omri374** [reply]
Here's an example:

```python
from presidio_analyzer import AnalyzerEngine

analyzer = AnalyzerEngine()
entities = analyzer.get_supported_entities()

entities_without_date = [ent for ent in entities if ent != "DATE_TIME"]

analyzer.analyze(text="They called me at 6pm", language = "en")
analyzer.analyze(text="They called me at 6pm", language = "en", entities = entities_without_date)
```

**Reply by omri374** [reply]
Hi, one option is to define recognizers via yaml, and only keep those that you need. Another option is to remove those that you don't need via code (like mentioned in this thread). Third is to start with an empty list and adding those that you need. For that, you can instantiate `RecognizerRegistry` and not call `load_predefined_recognizers`. There are examples to this in the samples section and across different issues answered.
